qiskit/ignis/experiments/calibration/instruction_data/compiler.py

```diff
@@ -0,0 +1,318 @@
+        self._gate_id = gate_id
+        self._free_parameters = free_parameters or []
+
+        return self.visit(parse(source))
+
+    def _get_channel(self, ch_str: str) -> pulse.channels.Channel:
+        """A helper function to convert channel string into Qiskit object.
+
+        Args:
+            ch_str: String representation of channel.
+
+        Returns:
+            Qiskit pulse channel object.
+        """
+        match = self.chan_regex.match(ch_str)
+        if match:
+            prefix, index = match.group(1), int(match.group(2))
+            channel = self._inst_def.pulse_channels[prefix].value
+            return channel(index=index)
+
+        raise Exception('Channel name {name} is not correct syntax.'.format(name=ch_str))
+
+    def visit(self, node):
+        """Visit a node."""
+        method = 'visit_' + node.__class__.__name__
+        visitor = getattr(self, method)
+
+        return visitor(node)
+
+    def visit_PulseInst(self, node: PulseInst) -> pulse.Instruction:
+        """Evaluate pulse instruction node and return play instruction.
+
+        Args:
+            node: PulseInst node to evaluate.
+
+        Returns:
+            Play instruction.
+        """
+        parametric_pulse = self._inst_def.parametric_shapes[node.shape].value
+
+        # generate parameter names on the fly
+        parametric_pulse_kwargs = dict()
+        for pname in utils.get_pulse_parameters(parametric_pulse):
+            scoped_pname = utils.add_scope(pname, node.channel, node.name)
+            # TODO calling parameter biding here is bit strange.
+            # This parameter binding should be offloaded to the get_gate_schedule method.
+            # However, we cannot do this now because of parametrization of duration.
+            # The role of parser is to create program, not the parameter-bound schedule.
+            param_val = self._inst_def.pulse_parameter_table.get_parameter(
+                parameter=scoped_pname,
+                gate_id=self._gate_id,
+                series=self._inst_def.series
+            )
+            if param_val is not None and scoped_pname not in self._free_parameters:
+                parametric_pulse_kwargs[pname] = param_val
+            else:
+                parametric_pulse_kwargs[pname] = circuit.Parameter(scoped_pname)
+
+        # pulse name for visualization purpose
+        pulse_name = '{}.{}.{}'.format(node.name, node.channel, self._gate_id)
+        played_pulse = parametric_pulse(**parametric_pulse_kwargs, name=pulse_name)
+        channel = self._get_channel(node.channel)
+
+        return pulse.Play(played_pulse, channel)
+
+    def visit_FrameInst(self, node: FrameInst) -> pulse.Instruction:
+        """Evaluate frame instruction node and return frame change instruction.
+
+        Args:
+            node: FrameInst node to evaluate.
+
+        Returns:
+            Arbitrary frame change instruction specified by AST name.
+        """
+        raise NotImplementedError
```

Comment:
Why not implemented?

Reply:
Because frame change syntax is not decided yet. Do we keep operand value in the program or store the value in parameter table?

Reply:
Right. If we keep it in the operand then we cannot calibrate it, correct? But if we have it in the parameter table then we can see it as a parameter to calibrate. Is it possible to support both options? Or would that be unpractical? I think in many gates there will be phases to calibrate. Could this be done solely with amplitude of the pulse?

Reply:
Currently this can be absorbed in the complex amplitude of the pulse. Sometime we may need a dedicated frame change to help understanding of schedule, i.e. CX = [IX]_-1/2@[ZI]_-1/2@[ZX]_1/2. In this case shift phase in the middle is kind of gate, but no need to be calibrated. 
X Y phase due to mixer skew can be absorbed by amplitude. I think a phase shift to calibrate is something like frame of composite gate, but I don't have any good example off the top of my head.